tests/sentry/rules/processing/test_delayed_processing.py

```diff
@@ -64,9 +122,121 @@ def test_bulk_fetch_events(self):
         pass
 
 
-class GetConditionGroupResultsTest(TestCase):
-    def test_get_condition_group_results(self):
-        pass
+class GetConditionGroupResultsTest(CreateEventTestCase):
+    interval = "1h"
+    comparison_interval = "15m"
+
+    def create_events(self, comparison_type: ComparisonType) -> GroupEvent:
+        # Create current events for the first query
+        event = self.create_event(self.project.id, FROZEN_TIME, "group-1", self.environment.name)
+        self.create_event(self.project.id, FROZEN_TIME, "group-1", self.environment.name)
+        if comparison_type == ComparisonType.PERCENT:
+            # Create a past event for the second query
+            self.create_event(
+                self.project.id,
+                FROZEN_TIME - timedelta(hours=1, minutes=10),
+                "group-1",
+                self.environment.name,
+            )
+        return event
+
+    def create_condition_groups(
+        self, condition_data_list: Sequence[EventFrequencyConditionData]
+    ) -> tuple[dict[UniqueConditionQuery, DataAndGroups], int, list[UniqueConditionQuery]]:
+        condition_groups = {}
+        all_unique_queries = []
+        for condition_data in condition_data_list:
+            unique_queries = generate_unique_queries(condition_data, self.environment.id)
+            event = self.create_events(condition_data["comparisonType"])
+            data_and_groups = DataAndGroups(
+                data=condition_data,
+                group_ids={event.group.id},
+            )
+            condition_groups.update({query: data_and_groups for query in unique_queries})
+            all_unique_queries.extend(unique_queries)
+        return condition_groups, event.group.id, all_unique_queries
+
+    def test_empty_condition_groups(self):
+        assert get_condition_group_results({}, self.project) == {}
+
+    @patch("sentry.rules.processing.delayed_processing.logger")
+    def test_nonexistent_condition_cls(self, mock_logger):
+        first_query = UniqueConditionQuery(cls_id="fake_id", interval="", environment_id=1)
+        fake_data_groups = DataAndGroups(
+            data=self.create_event_frequency_condition(id="fake_id"),
+            group_ids={1},
+        )
+        assert get_condition_group_results({first_query: fake_data_groups}, self.project) is None
```

Comment:
Do we want to return `None` for all conditions or just the specific condition? It looks like in code we will return `None` for all conditions if 1 condition doesn't exist.

My expectation is more that the this condition result would be none, not that _all_ condition results are none.

Reply:
oh yeah good catch, will fix this